src/features/MainScreen/NotesList/NoteContextMenu/useNoteContextMenu.ts

```diff
@@ -0,0 +1,188 @@
+import {
+	useNotesContext,
+	useNotesRegistry,
+	useTagsRegistry,
+} from '@features/App/Workspace/WorkspaceProvider';
+import { buildFileName, useNotesExport } from '@hooks/notes/useNotesExport';
+import { ContextMenuCallback } from '@hooks/useContextMenu';
+import { useShowContextMenu } from '@hooks/useShowContextMenu';
+import { useAppSelector } from '@state/redux/hooks';
+import { useWorkspaceData } from '@state/redux/profiles/hooks';
+import { selectWorkspace } from '@state/redux/profiles/profiles';
+import { copyTextToClipboard } from '@utils/clipboard';
+
+import { NoteActions } from '.';
+
+export type ContextMenuOptions = {
+	closeNote: (id: NoteId) => void;
+	updateNotes: () => void;
+};
+
+const mdCharsForEscape = ['\\', '[', ']'];
+const mdCharsForEscapeRegEx = new RegExp(
+	`(${mdCharsForEscape.map((char) => '\\' + char).join('|')})`,
+	'g',
+);
+
+const buildNoteMenu = (note: INote): ContextMenu => {
+	const commonActions: ContextMenu = [
+		{
+			id: NoteActions.EXPORT,
+			label: 'Export',
+		},
+		{
+			id: NoteActions.COPY_MARKDOWN_LINK,
+			label: 'Copy markdown link',
+		},
+		{ type: 'separator' },
+	];
+
+	return note.isDeleted
+		? [
+				{ id: NoteActions.RESTORE_FROM_BIN, label: 'Restore' },
+				...commonActions,
+				{ id: NoteActions.DELETE_PERMANENTLY, label: 'Delete permanently' },
+		  ]
+		: [
+				{ id: NoteActions.DUPLICATE, label: 'Duplicate' },
+				...commonActions,
+				{ id: NoteActions.DELETE_TO_BIN, label: 'Delete to bin' },
+		  ];
+};
+
+export const useNoteContextMenu = ({ closeNote, updateNotes }: ContextMenuOptions) => {
+	const notes = useNotesRegistry();
+	const tagsRegistry = useTagsRegistry();
+
+	const notesExport = useNotesExport();
+	const currentWorkspace = useWorkspaceData();
+	const workspaceData = useAppSelector(selectWorkspace(currentWorkspace));
+
+	const { noteUpdated } = useNotesContext();
+
+	const noteContextMenuCallback = useCallback<ContextMenuCallback<NoteActions>>(
+		async ({ id, action }) => {
+			const actionsMap = {
+				[NoteActions.DELETE_TO_BIN]: async (id: string) => {
+					const isConfirmed = confirm(
+						`Are you sure you want to move the note to the bin?`,
+					);
+					if (!isConfirmed) return;
+
+					closeNote(id);
+
+					await notes.updateMeta([id], { isDeleted: true });
+					const updatedNote = await notes.getById(id);
+					if (updatedNote) noteUpdated(updatedNote);
+
+					updateNotes();
+				},
+
+				[NoteActions.DELETE_PERMANENTLY]: async (id: string) => {
+					const isConfirmed = confirm(
+						`Are you sure you want to delete the note permanently?`,
+					);
+					if (!isConfirmed) return;
+
+					closeNote(id);
+
+					await notes.delete([id]);
+					await tagsRegistry.setAttachedTags(id, []);
+
+					updateNotes();
+				},
+
+				[NoteActions.RESTORE_FROM_BIN]: async (id: string) => {
+					const isConfirmed = confirm(
+						'Are you sure you want to restore the note',
+					);
+					if (!isConfirmed) return;
```

Comment:
Confirmations is required only for destructive actions.

Such behavior like this would slow down user significantly and affect UX very negative, since user have to click many buttons to do simple things.

That is very dangerous for UX.